kong/api/routes/upstreams.lua

```diff
@@ -131,59 +174,73 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      self.params.active = nil
+      local active, active_n = get_active_targets(dao_factory,
+                                                  self.params.upstream_id)
 
-      local target_history, err = dao_factory.targets:find_all({
-        upstream_id = self.params.upstream_id,
-      })
-      if not target_history then
-        return app_helpers.yield_error(err)
-      end
+      -- for now lets not worry about rolling our own pagination
+      -- we also end up returning a "backwards" list of targets because
+      -- of how we sorted- do we care?
+      return responses.send_HTTP_OK {
+        total = active_n,
+        data  = active,
+      }
+    end,
 
-      --sort and walk based on target and creation time
-      for _, target in ipairs(target_history) do
-        target.order = target.target .. ":" ..
-          target.created_at .. ":" .. target.id
-      end
-      table.sort(target_history, function(a, b) return a.order > b.order end)
+    POST = function(self, dao_factory, helpers)
+      clean_history(self.params.upstream_id, dao_factory)
 
-      local seen     = {}
-      local active   = setmetatable({}, cjson.empty_array_mt)
-      local active_n = 0
+      crud.post(self.params, dao_factory.targets)
+    end,
+  },
 
-      for _, entry in ipairs(target_history) do
-        if not seen[entry.target] then
-          if entry.weight == 0 then
-            seen[entry.target] = true
+  ["/upstreams/:upstream_name_or_id/health/"] = {
+    before = function(self, dao_factory, helpers)
+      crud.find_upstream_by_name_or_id(self, dao_factory, helpers)
+      self.params.upstream_id = self.upstream.id
+    end,
 
-          else
-            entry.order = nil -- dont show our order key to the client
+    GET = function(self, dao_factory)
+      local upstream_id = self.params.upstream_id
+      local active, active_n = get_active_targets(dao_factory, upstream_id)
 
-            -- add what we want to send to the client in our array
-            active_n = active_n + 1
-            active[active_n] = entry
+      local node_id, err = public.get_node_id()
+      if err then
+        ngx.log(ngx.ERR, "failed getting node id: ", err)
+      end
+
+      local health_info
+      health_info, err = balancer.get_upstream_health(upstream_id)
+      if err then
+        ngx.log(ngx.ERR, "[healthchecks] failed getting upstream health: ", err)
```

Comment:
`[healthchecks]` is the namespace we chose for the actual health-checker, but I am thinking maybe this one should be different since it is an error in the Admin API and not in the proxy itself (less worrying for users especially since no stacktrace is printed by `ngx.log`).

If we use an "Admin API" namespace we could also add it to the above error log (or no namespace at all would be fine to imho).